src/porepy/fracs/wells_3d.py

```diff
@@ -0,0 +1,583 @@
+
+            stop_ind = sort_inds.size - ignore_endpoint_tag
+            well_pts = np.hstack((well_pts, sorted_pts[:, :stop_ind]))
+            # The last tag might change when it is used for the start point of
+            # the next segment. Store remaining tags in correct order
+            [well_tags.append(tags_seg[i]) for i in sort_inds[:stop_ind]]
+        # Overwrite old points and tags for this well
+        well.p = well_pts
+        well.tags["intersecting_fractures"] = well_tags
+
+
+def _argsort_points_along_line_segment(
+    seg: np.ndarray,
+) -> Tuple[np.ndarray, np.ndarray]:
+    """Sort point lying along a segment.
+
+    Parameters:
+        seg (array): 3 x npts coordinates of the points to be sorted, assumed
+            to lie on a straight line.
+
+    Returns:
+        array (npts x 1): Indices of the sorting.
+        array (3 x npts): Sorted points.
+
+    The sorting is done so that
+    seg[d, inds[0]], seg[d,inds[1]], ..., seg[d,inds[-2]], seg[d,inds[-1]]
+    is monotone for at least one dimension d. Ascending or descending order
+    is determined by the values of the two end points.
+    """
+    for dim in range(3):
+        if not np.isclose(seg[dim, 0] - seg[dim, 1], 0):
+            break
+    inds = np.argsort(seg[dim])
+    if seg[dim, 0] > seg[dim, 1]:
+        inds = inds[::-1]
+    return inds, seg[:, inds]
+
+
+def _intersection_segment_fracture(
+    segment_points: np.ndarray,
+    fracture: pp.Fracture,
+    tags: List[np.ndarray],
+    ignore_endpoint_tag: bool,
+    tol: float = 1e-8,
+) -> Tuple[np.ndarray, List[np.ndarray]]:
+    """Compute intersection between a single line segment and fracture.
+
+    Checks whether the intersection corresponds to one of the endpoints and
+    provide that information.
+    segment_points are
+
+    Parameters:
+        segment_points (array, 3 x npts): coordinates of the points on the line segment,
+            sorted as [start, end, *any interior points]
+        fracture (pp.Fracture): the fracture to be checked for intersections
+            with the line segment.
+        tags (list of length npts): identifies fractures (by fracture.index)
+            intersecting at each of the points in segment_points.
+        ingore_endpoint_tag (bool): whether to update the tag of the second endpoint
+            or not. To be used when looping over a polyline. The last endpoint
+            of this segment will be treated as the first endpoint of the next
+            segment.
+        tol (float): Used to determine whether there is an intersection between
+            the segment and the fracture.
+
+    Returns:
+        segment_points (array, 3 x npts): updated coordinates of the points on
+            the line segment, sorted as [start, end, *any interior points].
+            Any new points have been appended.
+        tags (list of length npts): updated tags.
+    """
+    distance, isec_pt = pp.geometry.distances.segments_polygon(
```

Comment:
Some comments here, please, documenting the motivation for the compuation and the main steps in the method.